core/src/main/java/de/evoila/cf/broker/controller/BaseController.java

```diff
@@ -22,76 +22,76 @@ protected ResponseEntity processErrorResponse(HttpStatus status) {
-        return new ResponseEntity(new ResponseMessage(message), status);
+    protected ResponseEntity<ResponseMessage<String> > processErrorResponse(String message, HttpStatus status) {
+        return new ResponseEntity<>(new ResponseMessage<>(message), status);
     }
 
-    protected ResponseEntity processErrorResponse(String error, String description, HttpStatus status) {
+    protected ResponseEntity<ServiceBrokerErrorResponse> processErrorResponse(String error, String description, HttpStatus status) {
         log.debug("Handled following service broker error: " + error + " - " + description);
-        return new ResponseEntity(new ServiceBrokerErrorResponse(error, description), status);
+        return new ResponseEntity<>(new ServiceBrokerErrorResponse(error, description), status);
     }
 
     @ExceptionHandler({ValidationException.class})
-    public ResponseEntity<ResponseMessage> handleException(ValidationException ex) {
+    public ResponseEntity<ResponseMessage<String> > handleException(ValidationException ex) {
         return processErrorResponse(ex.getMessage(), HttpStatus.BAD_REQUEST);
     }
 
     @ExceptionHandler({MaintenanceInfoVersionsDontMatchException.class})
-    public ResponseEntity<ResponseMessage> handleException(MaintenanceInfoVersionsDontMatchException ex) {
+    public ResponseEntity<ServiceBrokerErrorResponse> handleException(MaintenanceInfoVersionsDontMatchException ex) {
         return processErrorResponse(ex.getError(), ex.getDescription(), HttpStatus.UNPROCESSABLE_ENTITY);
     }
 
     @ExceptionHandler({ConcurrencyErrorException.class})
-    public ResponseEntity<ResponseMessage> handleException(ConcurrencyErrorException ex) {
+    public ResponseEntity<ServiceBrokerErrorResponse> handleException(ConcurrencyErrorException ex) {
         return processErrorResponse(ex.getError(), ex.getDescription(), HttpStatus.UNPROCESSABLE_ENTITY);
     }
 
     @ExceptionHandler({AsyncRequiredException.class})
-    public ResponseEntity<ResponseMessage> handleException(AsyncRequiredException ex) {
+    public ResponseEntity<ServiceBrokerErrorResponse> handleException(AsyncRequiredException ex) {
         return processErrorResponse(ex.getError(), ex.getDescription(), HttpStatus.UNPROCESSABLE_ENTITY);
     }
 
     @ExceptionHandler(ServiceBrokerFeatureIsNotSupportedException.class)
-    public ResponseEntity<ResponseMessage> handleException(ServiceBrokerFeatureIsNotSupportedException ex) {
+    public ResponseEntity handleException(ServiceBrokerFeatureIsNotSupportedException ex) {
         return processErrorResponse(HttpStatus.UNPROCESSABLE_ENTITY);
     }
 
     @ExceptionHandler({ServiceDefinitionDoesNotExistException.class, ServiceDefinitionPlanDoesNotExistException.class, ServiceInstanceNotRetrievableException.class})
-    public ResponseEntity<ResponseMessage> handleException(Exception ex) {
+    public ResponseEntity<ResponseMessage<String> > handleException(Exception ex) {
```

Comment:
I think we should stay consistent here. As java is capable of deriving the type via `<>` and the `BaseController` is a generic approach of handling stuff, we should stay with either `ResponseEntity` or `ResponseEntity<>` and have it as standard for all methods.

Reply:
Change is from `ResponseEntity<ResponseMessage>`, not from `ResponseEntity`. Some of the `handleException` functions, e.g. for `MaintenanceInfoVersionsDontMatchException`, do not return `ResponseEntity<ResponseMessage>`. Example returns `ResponseEntity<ServiceBrokerErrorResponse>`. Change was made to remove unchecked assignment warnings and make the return type clear since `ResponseMessage` and `ServiceBrokerErrorResponse` can not be handled the same way. `<>` breaks the build. We can change to `ResponseEntity` but I don't think it's better. Is there no reason for the warnings?